pkg/ingester/ingester_v2.go

```diff
@@ -1126,18 +1151,28 @@ func (i *Ingester) compactBlocks(ctx context.Context) {
 		var err error
 
 		i.TSDBState.compactionsTriggered.Inc()
-		if i.cfg.TSDBConfig.HeadCompactionIdleTimeout > 0 && userDB.isIdle(time.Now(), i.cfg.TSDBConfig.HeadCompactionIdleTimeout) {
-			level.Debug(util.Logger).Log("msg", "Forcing compaction due to TSDB being idle")
+
+		reason := ""
+		switch {
+		case force:
+			reason = "forced"
 			err = userDB.CompactHead(tsdb.NewRangeHead(h, h.MinTime(), h.MaxTime()))
-		} else {
+
+		case i.cfg.TSDBConfig.HeadCompactionIdleTimeout > 0 && userDB.isIdle(time.Now(), i.cfg.TSDBConfig.HeadCompactionIdleTimeout):
+			reason = "idle"
+			level.Debug(util.Logger).Log("msg", "TSDB is idle, forcing compaction", "user", userID)
```

Comment:
I would log as Info, since it should be unfrequent but noticed. I missed this in the previous review when this was introduced.

Reply:
Sounds good. I was wondering whether we want to use `reason` as label for started/failed compaction counters? WDYT?